### PR TITLE
fuzz: remove invalid characters in header parser

### PR DESCRIPTION
**Pull request by asraa**
Filter through headers to add and remove in the header parser fuzzer.

Risk level: Low
Testing: Added corpus entry with null characters in headers to add.

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: test/common/router/header_parser_fuzz_test.cc [hunk 1]
```diff
@@ -9,11 +10,32 @@ namespace Envoy {
 namespace Fuzz {
 namespace {
 
+using Envoy::replaceInvalidCharacters;
+
+// Remove invalid characters from HeaderValueOptions in headers to add.
+Protobuf::RepeatedPtrField<envoy::api::v2::core::HeaderValueOption> processHeadersToAdd(
```
Comment (htuch) [inline test/common/router/header_parser_fuzz_test.cc]: Should this helper be in https://github.com/envoyproxy/envoy/blob/master/test/fuzz/utility.h? That is ideally where we can centralize the `Header` message style utils.

Comment (asraa) [reply test/common/router/header_parser_fuzz_test.cc]: Yes!! I didn't see that. Honestly, I think I should move the function in https://github.com/envoyproxy/envoy/blob/master/test/common/http/common_fuzz.h here as well.

Comment (htuch) [reply test/common/router/header_parser_fuzz_test.cc]: Sure, makes sense.

File: test/common/http/header_map_impl_fuzz_test.cc [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
 #include "common/http/header_map_impl.h"
 
-#include "test/common/http/common_fuzz.h"
 #include "test/common/http/header_map_impl_fuzz.pb.h"
 #include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
 
 namespace Envoy {
 
+using Fuzz::replaceInvalidCharacters;
```
Comment (htuch) [inline test/common/http/header_map_impl_fuzz_test.cc]: Nit: we typically place `using` outside the namespace scopes in Envoy code.

File: test/fuzz/utility.h [hunk 3]
```diff
@@ -76,5 +76,45 @@ inline TestStreamInfo fromStreamInfo(const test::fuzz::StreamInfo& stream_info)
...
+      break;
+    default:
+      filtered.push_back(c);
+    }
+  }
+  return filtered;
+}
+
+// Return a new RepeatedPtrField of HeaderValueOptions with invalid characters removed.
+inline Protobuf::RepeatedPtrField<envoy::api::v2::core::HeaderValueOption> processHeadersToAdd(
```
Comment (htuch) [inline test/fuzz/utility.h]: Nit: prefer something more descriptive like `replaceInvalidHeaders` (or whatever you prefer)

**Pull request merged**